packages/encrypted-blockstore/src/loader.ts

```diff
@@ -241,19 +269,29 @@ export class Loader implements Loadable {
     opts: CommitOpts = { noLoader: false, compact: false }
   ): Promise<CarGroup> {
     await this.ready
-    const fp = this.makeCarHeader(done, this.carLog, !!opts.compact) as CarHeader
+    const fp = this.makeCarHeader(
+      done,
+      this.carLog,
+      !!opts.compact
+    ) as CarHeader
     let roots: AnyLink[] = await this.prepareRoots(fp, t)
     //We need to split the data inside the prepareCarFile?
     //While splitting every CAR file should have a copy of root[0] meaning it should have a copy of fp
     // Maximum size of each CAR file as 1mb?
-    const { cid, bytes } = await this.prepareCarFile(roots[0], t, !!opts.public)
-    await this.carStore!.save({ cid, bytes })
-    await this.cacheTransaction(t)
-    const newDbMeta = { cars: [cid], key: this.key || null } as DbMeta
-    await this.remoteWAL!.enqueue(newDbMeta, opts)
-    await this.metaStore!.save(newDbMeta)
-    await this.updateCarLog([cid], fp, !!opts.compact)
-    return [cid]
+    const cars = await this.prepareCarFiles(roots, t, !!opts.public)
+    const cids: AnyLink[] = []
+    for (const car of cars) {
+      const { cid, bytes } = car
+      await this.carStore!.save({ cid, bytes })
+      await this.cacheTransaction(t)
+      const newDbMeta = { cars: [cid], key: this.key || null } as DbMeta
+      await this.remoteWAL!.enqueue(newDbMeta, opts)
+      await this.metaStore!.save(newDbMeta)
+      await this.updateCarLog([cid], fp, !!opts.compact)
+      cids.push(cid)
```

Comment:
same with the other lines:

```
const newDbMeta = { cars: [cid], key: this.key || null } as DbMeta
       await this.remoteWAL!.enqueue(newDbMeta, opts)
       await this.metaStore!.save(newDbMeta)
       await this.updateCarLog([cid], fp, !!opts.compact)
```

they should happen after the loop ends and the cids are pushed.